File: tags/rl-glue.2.0.2/Tests/Network_C/src/experiment.c
#include <stdio.h>
#include <RL_glue.h>

#define NUM_EPISODES 1
int rl_num_steps[NUM_EPISODES];
double rl_return[NUM_EPISODES];

void run(int num_episodes) {        
  int x = 0;
  for(x = 0; x < num_episodes; ++x) {
    RL_episode(0);
    fprintf(stderr, ".");
    rl_num_steps[x] = RL_num_steps();
    rl_return[x] = RL_return();
  }
}

int main(int argc, char *argv[]) {
  int trial = 0;
  unsigned int i = 0;
  double avg_steps = 0.0;
  double avg_return = 0.0;

  char* agent_response = 0;
  char* env_response = 0;

  for (trial = 0; trial < 1; ++trial) {

    agent_response = RL_agent_message(0);
    env_response = RL_env_message(0);

    fprintf(stderr, "agent_response: |%s|\n", agent_response);
    fprintf(stderr, "env_response: |%s|\n", env_response);

    RL_init();
    run(NUM_EPISODES);
    RL_cleanup();
    for (i = 0; i < NUM_EPISODES; i++) {
      avg_steps += rl_num_steps[i];
      avg_return += rl_return[i];
    }
    
    avg_steps /= NUM_EPISODES;
    avg_return /= NUM_EPISODES;
    printf("\n-----------------------------------------------\n");
    printf("Number of episodes: %d\n",NUM_EPISODES);
    printf("Average number of steps per episode: %f\n", avg_steps);
    printf("Average return per episode: %f\n", avg_return);
    printf("-----------------------------------------------\n");
  }
   
  return 0;
}

File: Examples/Network_Java/src/Experiment.java
import rlglue.RLGlue;
import rlglue.types.*;
import java.io.IOException;

public class Experiment
{
	protected static final int kNumEpisodes = 100;
	protected static int rlNumSteps[];
	protected static double rlReturn[];

	protected static void run(int numEpisodes) throws IOException
	{
		/*run for num_episode number of episodes and store the number of steps and return from each episode*/        	
		for(int x = 0; x < numEpisodes; ++x) {
			RLGlue.RL_episode(0);
			System.out.print(".");
			rlNumSteps[x] = RLGlue.RL_num_steps();
			rlReturn[x] = RLGlue.RL_return();
		}
	}

	public static void main(String [] args) throws IOException {
		double avgSteps = 0.0;
		double avgReturn = 0.0;

		rlNumSteps = new int[Experiment.kNumEpisodes];
		rlReturn = new double[Experiment.kNumEpisodes];

		RLGlue.RL_init();

		String agentMessage = RLGlue.RL_agent_message("poke");
		System.out.println(agentMessage);
		
		String msg = "poke";
		
		while (msg.length() < 65536) {
		    msg += "0123456789" + msg;
		}
				
		String envMessage = RLGlue.RL_env_message(msg);
		System.err.println(msg);
		
		run(kNumEpisodes);
		RLGlue.RL_cleanup();
		
		/*add up all the steps and all the returns*/
		for (int i = 0; i < Experiment.kNumEpisodes; ++i) {
		    avgSteps += rlNumSteps[i];
		    avgReturn += rlReturn[i];
		}
		
		/*average steps and returns*/
		avgSteps /= (double)Experiment.kNumEpisodes;
		avgReturn /= (double)Experiment.kNumEpisodes;
		
		/*print out results*/
		System.out.println("\n-----------------------------------------------\n");
		System.out.println("Number of episodes: " + Experiment.kNumEpisodes);
		System.out.println("Average number of steps per episode: " +  avgSteps);
		System.out.println("Average return per episode: " + avgReturn);
		System.out.println("-----------------------------------------------\n");
	}   
}
